src/map/mob.cpp

```diff
@@ -4283,7 +4283,8 @@ uint64 MobDatabase::parseBodyNode(const YAML::Node &node) {
 			return 0;
 
 		mob = std::make_shared<s_mob_db>();
-		mob->vd.class_ = static_cast<uint16>(mob_id);
+		mob->id = mob_id;
+		mob->vd.class_ = static_cast<uint16>(mob->id);
```

Comment:
we should add a secury check for mob_id > UINT16_MAX here.

you have set mob->id to uint32 which is the same as mob_id. but vd.class_ is a uint16. if there is ever the case where mob_id exceeds UINT16_MAX, mob->id would be fine, but vd.class_ would be truncated without any notice.

Reply:
Hey!
Thank you for your (valid) input.
I did not bother to add/adjust any checks that were not there before.
I agree with you that code-wise it looks bad to downcast it like this, but for the time being we should be out of danger, because we can assume `#define MAX_MOB_DB2 31999`. Once this limit is lifted from Gravity clientside, this will be a whole different story though (just like with item ids).